### ISSUE TITLE
system font error on `theme_hdx()`

### ISSUE THREAD

**Issue by zackarno** [OCHA-DAP/gghdx#11]
gghdx was working great on my Windows, but just tried it on a mac and it looks like there is a font related error being thrown from `theme_hdx()` ?

I'm guessing there is some system fonts package I had installed on the other computer, but not this one? If so, perhaps it should be a dependency?

reprex below:

``` r
library(tidyverse)
library(gghdx)

# Create data
age <- c(25, 30, 35, 40, 45, 50, 55, 60, 65, 70)
gender <- c("male", "female", "male", "female", "male", "female", "male", "female", "male", "female")

# Create data frame
example_data <- data.frame(age, gender)

example_data %>% 
  ggplot(aes(x=gender,y=age))+
  geom_boxplot() +
  theme_hdx()
#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"

#> Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): no font could
#> be found for family "Source Sans Pro"
#> Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : no
#> font could be found for family "Source Sans Pro"
#> Error in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : polygon edge not found

warnings()
```

<sup>Created on 2023-01-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by zackarno** [comment]
hmm i got it working by restarting my computer and restarting R several times and then including `font_add_google()` and `showtext_auto()` (reprex below). I saw that you ran these in one of the vignettes, I guess the user is supposed to run these before running the gghdx funcs?

``` r
library(tidyverse)
library(gghdx)
library(sysfonts)
font_add_google("Source Sans Pro")
showtext::showtext_auto()


# Create data
age <- c(25, 30, 35, 40, 45, 50, 55, 60, 65, 70)
gender <- c("male", "female", "male", "female", "male", "female", "male", "female", "male", "female")
example_data <- data.frame(age, gender)

example_data %>% 
  ggplot(aes(x=gender,y=age))+
  geom_boxplot() +
  theme_hdx()
```

![](https://i.imgur.com/vPyNYRH.png)

<sup>Created on 2023-01-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by caldwellst** [comment]
Okay yes, I don't remember but probably th e case! Can put in some warning/error catching. Will check in a bit!

**Comment by zackarno** [comment]
this one got me again! The above example no longer works (same computer)... even with the addition of

```
library(sysfonts)
font_add_google("Source Sans Pro")
showtext::showtext_auto()
```
perhaps this info will help trouble shooting.


seems to be related to the fonts. I wonder if different default font might help?

**Comment by zackarno** [comment]
> this one got me again! The above example no longer works (same computer)- same error as above ... even with the addition of
> 
> ```
> library(sysfonts)
> font_add_google("Source Sans Pro")
> showtext::showtext_auto()
> ```
> 
> perhaps this info will help trouble shooting.
> 
> seems to be related to the fonts. I wonder if different default font might help?

Wait... restarting R 2x made the above solution work - that's odd

**Comment by caldwellst** [comment]
Fixed in #13
